Docker and OpenAI API Call

manorajesh opened the pull request:
- Added `Dockerfile` for easy development
- Updated `BUILDING.md` to explain the current Docker setup and deployment
- Backend making a POST request to the `gpt3.5-turbo` model for idea generation
- Custom prompts/criteria supported for generation
- API key assumed to be in an environment variable

sahendrickson commented on BUILDING.md:
```diff
@@ -178,3 +178,30 @@ npm start
 
 [^1]: Oxford English Dictionary
+
+## Running Project Leo in a Docker Container
```
How about making this as a third option under "## Running Project Leo"?

- Change "There are two ways to run Project Leo:" to "Project Leo can be run in the following ways:"
- Move this section before "[^1] ...", that should stay at the bottom of the page.
- Change the header from ## to ###.

sahendrickson commented on Dockerfile:
```diff
@@ -0,0 +1,20 @@
+FROM ubuntu:22.04
+
+# Install Apache Maven, Git, Java, Nodejs, and Sass.
+RUN apt update && apt install -y maven git openjdk-17-doc openjdk-17-jdk openjdk-17-source nodejs sass
```
Can you try this without "nodejs" and "sass"? I think "mvn verify" will still work.

sahendrickson reviewed (commented):
One comment to make your section fit a little better in the documentation and another to see if you can remove some apt install dependencies.

manorajesh commented:
> One comment to make your section fit a little better in the documentation and another to see if you can remove some apt install dependencies.

Addressed issues

manorajesh commented:
Should there be some form of logging for `OpenAIController.java`? I kept getting errors when I added a logger which was set up similarly to the one in `ReactResourceController.java`. In the `server/pom.xml`, a comment read, "Log4j only allows one logger implementation to be registered." Can there only be one logger active at once?

sahendrickson commented:
I'm not sure what's going on with the logger. I may have set it up incorrectly. Let's get what you have in and then we'll figure out those things.
